src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addBylineBlock/index.js

```diff
@@ -0,0 +1,48 @@
+import pathOr from 'ramda/src/pathOr';
+import path from 'ramda/src/path';
+import deepClone from 'ramda/src/clone';
+import convertToBylineBlock from './models';
+
+const getBylineBlock = json => {
+  const byline = pathOr(null, ['promo', 'byline'], json);
+
+  if (!byline) {
+    return null;
+  }
+
+  return {
+    name: path(['name'], byline) || path(['persons', '0', 'name'], byline),
+    title:
+      path(['title'], byline) || path(['persons', '0', 'function'], byline),
+  };
+};
+
+const addBylineBlock = originalJson => {
+  const json = deepClone(originalJson);
+  const pageType = pathOr(null, ['metadata', 'type'], json);
+
+  const blocks = pathOr(null, ['content', 'model', 'blocks'], json);
+
+  if (!blocks || pageType !== 'STY') {
+    return json;
+  }
+
+  const byline = getBylineBlock(json);
+  if (byline) {
+    const bylineBlock = convertToBylineBlock(byline);
+
+    const headline = blocks.find(({ type }) =>
+      ['headline', 'fauxHeadline'].includes(type),
+    );
+
+    const remainingBlocks = blocks.filter(
+      ({ type }) => !['headline', 'fauxHeadline'].includes(type),
+    );
```

Comment:
We have similar logic here: https://github.com/bbc/simorgh/blob/07e44a4a401e06658ea067d42992d60f8b3c8c7b/src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock/index.js#L31..L35 

Should be consider extracting this out as a seperate module for finding heading? Also allows us to use 'find references' in VS code to see how many places we are dropping some content under the headline.

Reply:
@andrewscfc that makes sense. where do you think would be a good place to add that? I had considered creating a `helpers` folder in  `src/app/lib/utilities/preprocessor/rules/cpsAssetPage/` directory and adding commonly used logic across the rules there.

Reply:
Sounds good to me that location 🙂